docs/dictionary.rst

```diff
@@ -0,0 +1,1101 @@
+
+Further reading: `<https://docker-curriculum.com/>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Docker container:
+
+Docker container
+----------------
+	In order to actually use the software inside a :ref:`dict Docker image` using the `docker run` command, the Docker program creates a writable :ref:`dict layer` on top of the image, which leads to the creation of a :ref:`dict Docker container`. You can think of a Docker image as an unchanging template, and a Docker container as a writable instance generated from that template. A Docker image can exist on its own, but a Docker container requires a Docker image.  
+
+Further reading: `<https://www.docker.com/resources/what-container/>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Docker image:
+
+Docker image
+------------
+	A read-only file that represents a filesystem that contains some sort of code and that code's dependencies. A Docker image can be created using the `docker build` command in conjunction with a :ref:`dict Dockerfile`. If a workflow language references a Docker image, then the workflow executor will download that Docker image (unless was already downloaded previously) and add a writable layer onto the Docker image, which results in the creation of a :ref:`dict Docker container`.  
+
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Dockerfile:
+
+Dockerfile
+----------
+	A file describing the creation of a :ref:`dict Docker image` by running commands that each form a :ref:`dict layer`.  
+
+Further reading: `<https://docs.docker.com/engine/reference/builder/>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Dockstore CLI:
+
+Dockstore CLI
+-------------
+*abbreviation for* Dockstore Command Line Interface  
+
+	A command-line program developed by Dockstore. It is not required to use Dockstore, but it has many features to make running and developing workflows easier.  
+
+see also :ref:`dict CLI`  
+
+Further reading: `<https://docs.dockstore.org/en/stable/advanced-topics/dockstore-cli/dockstore-cli-faq.html>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Dockstore GitHub App:
+
+Dockstore GitHub App
+--------------------
+	The GitHub App that allows for Dockstore to communicate with GitHub repositories.  
+
+see also :ref:`dict GitHub App registration`  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict DOI:
+
+DOI
+---
+*abbreviation for* Digital Object Identifier  
+
+	An identifier that provides a long-lasting link to some sort of :ref:`dict immutable` digital object. On Dockstore, you can use Zenodo to mint a DOI of your workflows and tools to increase reproducibility.  
```

Comment:
Should elaborate that DOIs are useful for citing programs in publications. 
Software should be cited like all other research products, etc 
https://cite.research-software.org/